kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaServiceConfiguration.java

```diff
@@ -477,6 +477,13 @@ private String checkAdvertisedListeners(String advertisedListeners) {
                 } catch (UnknownHostException e) {
                     throw new IllegalStateException("hostname is empty and localhost is unknown: " + e.getMessage());
                 }
+            } else if (hostname.equalsIgnoreCase("advertisedAddress")) {
```

Comment:
If we treat `advertisedAddress` as a special hostname, it's better to use `equals` instead of `equalsIgnoreCase` to avoid a lot of (2^16) candidate hostnames are treated as the special hostname.